bodo/utils/conversion.py

```diff
@@ -2245,6 +2245,25 @@ def overload_index_to_array(I):
     return lambda I: bodo.hiframes.pd_index_ext.get_index_data(I)  # pragma: no cover
 
 
+def index_to_array_list(data):  # pragma: no cover
+    return data
+
+
+@overload(index_to_array_list, no_unliteral=True, jit_options={"cache": True})
+def overload_index_to_array_list(I):
+    """
+    convert Index object to data array.
+    """
+    from bodo.hiframes.pd_multi_index_ext import MultiIndexType
+
+    if isinstance(I, MultiIndexType):
+        return lambda I: list(
+            bodo.hiframes.pd_index_ext.get_index_data(I)
+        )  # pragma: no cover
+
+    return lambda I: [bodo.utils.conversion.index_to_array(I)]  # pragma: no cover
```

Comment:
For MultiIndex, `index_to_array` will return a tuple of arrays while the others return an array. So I created this helper function to make the output type consistent across all types.

Reply:
But array types can be different in MultiIndex so list wouldn't work. Am I missing something? Why not tuple?

Reply:
Hmm so the test case does use a MultiIndex with different types so it is working correctly. Looking further, I think whats happening is `bodo.hiframes.pd_index_ext.get_index_data` returns a tuple for MultiIndex so the output is either a tuple or list (with a single elem). It works fine because tuples and lists are similar enough. But you're right, to be safe I will change it to always return a tuple.

Reply:
Oh never mind I was looking at something else.

Reply:
Updated